monai/data/image_reader.py

```diff
@@ -766,7 +766,9 @@ def _get_seg_data(self, img):
             all_segs = np.zeros([*spatial_shape, n_classes])
 
         for i, (frames, description) in enumerate(self._get_frame_data(img)):
-            class_name = description.SegmentDescription
+            class_name = (
+                description.SegmentLabel if hasattr(description, "SegmentLabel") else description.SegmentDescription
+            )
```

Comment:
Still has the potential issue of AttributeError: 'Dataset' object has no attribute 'SegmentDescription'?